tests/test_intercept.js

```diff
@@ -921,6 +922,31 @@ test('reply with date header', async t => {
   scope.done()
 })
 
+// async / got version is returning "not ok test unfinished".
+// https://github.com/nock/nock/issues/1305#issuecomment-451701657
+test('reply with implicit date header', t => {
+  const clock = lolex.install()
+  const date = new Date()
+
+  const scope = nock('http://example.test')
+    .replyDate()
+    .get('/')
+    .reply(200)
+
+  mikealRequest.get('http://example.test', (err, resp) => {
+    clock.uninstall()
+
+    if (err) {
+      throw err
+    }
+
+    t.equal(resp.headers.date, date.toUTCString())
+    scope.done()
+
+    t.end()
+  })
+})
```

Comment:
just that I understand, it got stock with `got`, but works with `mikealRequest`?

Reply:
This looks good by the way, merge if you are done with it

Reply:
Yes, exactly. I'm not sure what to make of it.